app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListAdapter.java

```diff
@@ -35,15 +40,23 @@ public ContributionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int
 
     @Override
     public void onBindViewHolder(@NonNull ContributionViewHolder holder, int position) {
-        final Contribution contribution = callback.getContributionForPosition(position);
+        final Contribution contribution = contributions.get(position);
         DisplayableContribution displayableContribution = new DisplayableContribution(contribution,
                 position);
         holder.init(position, displayableContribution);
     }
 
     @Override
     public int getItemCount() {
-        return callback.getNumberOfContributions();
+        return contributions==null?0:contributions.size();
```

Comment:
how could contributions be null? Create it at the field level if you want. It should also be final.

There are minor style violations in this file, no spaces around operators

Reply:
Done